src/setup_payload/QRCodeSetupPayloadGenerator.cpp

```diff
@@ -202,16 +205,33 @@ CHIP_ERROR QRCodeSetupPayloadGenerator::payloadBase38Representation(std::string
                                                                     uint32_t tlvDataStartSize)
 {
     size_t tlvDataLengthInBytes = 0;
+    std::vector<uint8_t> bits(kTotalPayloadDataSizeInBytes);
+    MutableByteSpan bitsSpan;
+    std::vector<char> buffer;
+    MutableCharSpan bufferSpan;
 
     VerifyOrReturnError(mPayload.isValidQRCodePayload(), CHIP_ERROR_INVALID_ARGUMENT);
     ReturnErrorOnFailure(generateTLVFromOptionalData(mPayload, tlvDataStart, tlvDataStartSize, tlvDataLengthInBytes));
 
-    return payloadBase38RepresentationWithTLV(mPayload, base38Representation, kTotalPayloadDataSizeInBytes + tlvDataLengthInBytes,
-                                              tlvDataStart, tlvDataLengthInBytes);
+    bits.reserve(kTotalPayloadDataSizeInBytes + tlvDataLengthInBytes);
+    bitsSpan = MutableByteSpan(bits.data(), bits.capacity());
+    buffer.reserve(base38EncodedLength(bits.capacity()) + strlen(kQRCodePrefix));
+    bufferSpan = MutableCharSpan(buffer.data(), buffer.capacity());
+
+    ReturnErrorOnFailure(payloadBase38RepresentationWithTLV(mPayload, bufferSpan, bitsSpan, tlvDataStart, tlvDataLengthInBytes));
+
+    base38Representation.assign(bufferSpan.data());
+    return CHIP_NO_ERROR;
 }
 
-#if !defined(__clang__)
-#pragma GCC diagnostic pop // -Wstack-usage
-#endif
+CHIP_ERROR QRCodeSetupPayloadGenerator::payloadBase38RepresentationWithoutOptional(MutableCharSpan & outBuffer)
+{
+    // Do not call overloaded functions from here. Smaller devices that don't
+    // support std::string must avoid paths that bring in optional TLV data.
```

Comment:
How would this file compile at all if `std::string` is not supported?  Should the parts that use it be inside some sort of ifdefs?

Reply:
Constrained embedded devices generally provide stubs for unsupported dependencies in third-party source code. Although I agree, some builds may need to add ifdefs. I will leave that for a future PR.